### ISSUE TITLE
How to sync selected repositories / groups on Gitlab

### ISSUE THREAD

**Issue by illagrenan** [sourcebot-dev/sourcebot#36]
Thanks for a great project. I'm trying to sync git repositories with self-hosted Gitlab and would like to sync only repositories from a specific group:

```
https://gitlab.example.com/my-group/my-project/
```

This is my configuration:

```
{
  "$schema": "https://raw.githubusercontent.com/sourcebot-dev/sourcebot/main/schemas/index.json",
  "configs": [
    {
      "type": "gitlab",
      "gitlabURL": "https://gitlab.example.com",
      "Name": "^(my-group/.*|my-other-group/.*)$"
    }
  ]
}
```

In the log, I see that zoekt-mirror-gitlab is getting the configuration like this:

```
run [zoekt-mirror-gitlab -dest /data/.sourcebot/repos -url https://gitlab.example.com -name ^(my-group/.*|my-other-group/.*)$]
```

And yet, no repositories are synchronized. I'd say I have the regex wrong, but I haven't found any additional info on this.

Thanks in advance for the help!

**Comment by brendan-kellam** [comment]
Hey thanks for the issue! I looked into this and it's a pretty subtle bug. When performing the filtering in `zoekt-mirror-gitlab`, zoekt is using the [name_with_namespace](https://github.com/sourcegraph/zoekt/blob/5687809315075882a8e7413bdb17b042f3394c02/cmd/zoekt-mirror-gitlab/main.go#L129) field of the project to filter on. [Looking at the GitLab docs](https://docs.gitlab.com/search/?q=name_with_namespace&page=1) reveals that `name_with_namespace` has spaces between group, `/`, and the repository:

<img width="447" alt="image" src="https://github.com/user-attachments/assets/c148df0b-9394-4659-97a0-df9d67acfe6a">


The regex is therefore not matching. I validated that using `^(my-group /.*|my-other-group /.*)$` (notice the space) worked.

What we really want to use is `path_with_namespace` so we don't have to deal with this. Will make a fix now.

**Comment by brendan-kellam** [comment]
Just released [v1.0.2](https://github.com/sourcebot-dev/sourcebot/releases/tag/v1.0.2) with the fix - let me know if it fixed it for you!

**Comment by illagrenan** [comment]
Thank you so much for such a quick fix. Everything seems to be working as it should. gj!
